src/components/overlay/overlay.ts

```diff
@@ -0,0 +1,376 @@
+ * @cssprop [--sbb-overlay-z-index=var(--sbb-overlay-z-index)] - To specify a custom stack order,
+ * the `z-index` can be overridden by defining this CSS variable. The default `z-index` of the
+ * component is set to `var(--sbb-overlay-z-index)` with a value of `1000`.
+ */
+@customElement('sbb-overlay')
+export class SbbOverlayElement extends SbbNegativeMixin(LitElement) {
+  public static override styles: CSSResultGroup = style;
+  public static readonly events = {
+    willOpen: 'willOpen',
+    didOpen: 'didOpen',
+    willClose: 'willClose',
+    didClose: 'didClose',
+    backClick: 'requestBackAction',
+  } as const;
+
+  /**
+   * Whether to allow the overlay content to stretch to full width.
+   * By default, the content has the appropriate page size.
+   */
+  @property({ reflect: true, type: Boolean }) public expanded = false;
+
+  /**
+   * Whether a back button is displayed next to the title.
+   */
+  @property({ attribute: 'back-button', type: Boolean }) public backButton = false;
+
+  /**
+   * This will be forwarded as aria-label to the close button element.
+   */
+  @property({ attribute: 'accessibility-close-label' }) public accessibilityCloseLabel:
+    | string
+    | undefined;
+
+  /**
+   * This will be forwarded as aria-label to the back button element.
+   */
+  @property({ attribute: 'accessibility-back-label' }) public accessibilityBackLabel:
+    | string
+    | undefined;
+
+  /**
+   * This will be forwarded as aria-label to the relevant nested element.
+   */
+  @property({ attribute: 'accessibility-label' }) public accessibilityLabel: string | undefined;
+
+  /**
+   * Whether the animation is enabled.
+   */
+  @property({ attribute: 'disable-animation', reflect: true, type: Boolean })
+  public disableAnimation = false;
+
+  /*
+   * The state of the overlay.
+   */
+  private set _state(state: SbbOverlayState) {
+    this.dataset.state = state;
+  }
+  private get _state(): SbbOverlayState {
+    return this.dataset?.state as SbbOverlayState;
+  }
+
+  private _ariaLiveRef!: HTMLElement;
+  private _ariaLiveRefToggle = false;
+
+  /** Emits whenever the `sbb-overlay` starts the opening transition. */
+  private _willOpen: EventEmitter<void> = new EventEmitter(this, SbbOverlayElement.events.willOpen);
+
+  /** Emits whenever the `sbb-overlay` is opened. */
+  private _didOpen: EventEmitter<void> = new EventEmitter(this, SbbOverlayElement.events.didOpen);
+
+  /** Emits whenever the `sbb-overlay` begins the closing transition. */
+  private _willClose: EventEmitter = new EventEmitter(this, SbbOverlayElement.events.willClose);
+
+  /** Emits whenever the `sbb-overlay` is closed. */
+  private _didClose: EventEmitter<CloseEventDetails> = new EventEmitter(
+    this,
+    SbbOverlayElement.events.didClose,
+  );
+
+  /** Emits whenever the back button is clicked. */
+  private _backClick: EventEmitter<any> = new EventEmitter(
```

Comment:
Are you sure that the overlay should have a backButton?

Reply:
I think it should, it had it before in the the dialog `full-screen` mode and I see it also in Figma.